gadgets/trace_dns/README.mdx

```diff
@@ -19,52 +19,165 @@ Running the gadget:
+        :::
+
+        The overall flow of DNS request is as follows:
+
+        ```mermaid
+        sequenceDiagram
+        participant A as Application Pods
+        participant B as kube-dns Service
+        participant C as CoreDNS Pod
+        participant D as Upstream DNS
+
+        A->>B: DNS Request<br/> e.g Kubernertes Service / External name
+        B->>C: Forward Request<br/> e.g forwarding by host iptables
+        C->>D: Resolve or Forward Request<br/> e.g forwarding if external name
+        D-->>C: Response
+        C-->>B: Response
+        B-->>A: Response
+        ```
+
+        ### Tracing an Application Pod:
+
+        Let's try to understand the flow at different stages. We will start by tracing DNS requests for specific pod. First, create a namespace:
 
         ```bash
-        $ kubectl create namespace demo
+        kubectl create namespace demo
+        ```
+
+        with expected output:
+
+        ```
         namespace/demo created
         ```
 
-        Run the gadget in a terminal:
+        Run the gadget in a terminal, in this case we are only focused on DNS requests/response in `demo` namespace for pod `mypod`:
 
         ```bash
-        $ kubectl gadget run trace_dns:latest --namespace demo
-        K8S.NODE             K8S.NAMESPACE        K8S.PODNAME          K8S.CONTAINERNAME    SRC                         DST                         COMM              PID QR QTYPE      NAME                RCODE    ADDRESSES
+        kubectl gadget run trace_dns:latest -n demo -p mypod
         ```
 
-        Run a pod on a different terminal and perform some DNS requests:
+        Run a pod in a different terminal:
 
         ```bash
-        $ kubectl -n demo run mypod -it --image=wbitt/network-multitool -- /bin/sh
-        # nslookup -querytype=a inspektor-gadget.io.
-        # nslookup -querytype=aaaa inspektor-gadget.io.
-        # nslookup -querytype=mx inspektor-gadget.io.
+        kubectl -n demo run mypod -it --image=wbitt/network-multitool -- /bin/sh
         ```
+        Inside the pod, perform a DNS request:
+
+        ```sh
+        nslookup -query=a inspektor-gadget.io.
+        ```
+
+        The request/response will be logged by the DNS gadget:
+
+        ```
+        K8S.NODE             K8S.NAMESPACE        K8S.PODNAME          K8S.CONTAINERNAME    SRC                       DST                       COMM              PID QR QTYPE      NAME                RCODE    ADDRESSES
+        minikube-docker      demo                 mypod                mypod                10.244.0.4:56093          10.96.0.10:53             isc-net-0…     422204 Q  A          inspektor-gadget.i…
+        minikube-docker      demo                 mypod                mypod                10.96.0.10:53             10.244.0.4:56093          isc-net-0…     422204 R  A          inspektor-gadget.i… Success  172.67.16…
```

Comment:
Could we have an output where the name and addresses are not truncated?

Why are src/dst not enriched?

Reply:
IIUC, [kubeipresolver](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/operators/kubemanager/kubemanager.go) hasn't been adopted to work for image-based gadgets unless am I missing something. @flyth or @mauriciovasquezbernal do you have any thoughts?

Reply:
I documented it by mistake on #3406 but it doesn't work with image-based gadgets yet.